src/exports.cc

```diff
@@ -236,10 +236,11 @@ Word call_foreign_function(void *raw_context, Word function_name, Word function_
     result_size = s;
     return result;
   });
-  if (!context->wasmVm()->setWord(results, Word(address))) {
+  if ((address || results) && !context->wasmVm()->setWord(results, Word(address))) {
```

Comment:
Shouldn't this check only for `results`? i.e. if the caller provided address where it expects results to be written, we should always write them there, even if we write `0`, since we don't know what's currently written in that memory address.

Reply:
Makes sense.   Done. and below.